### PR TITLE
Fix: Do not log personal data in log messages

### PR DESCRIPTION
**Pull request by sanderhollaar**

### CODE REVIEW

File: src/main/java/foundation/privacybydesign/email/EmailRestApi.java [hunk 1]
```diff
@@ -125,7 +125,7 @@ public Response sendEmailToken(@FormParam("email") String emailAddress,
                     conf.getServerURL(language) + "#verify-email/" + token
             );
         } catch (AddressException e) {
-            logger.error("Invalid address: {}: {}", emailAddress, e.getMessage());
+            logger.error("Invalid address: {}", e.getMessage());
```
Comment (ivard) [inline src/main/java/foundation/privacybydesign/email/EmailRestApi.java:128]: The same here.

File: src/main/java/foundation/privacybydesign/email/EmailSender.java [hunk 2]
```diff
@@ -79,7 +79,7 @@ protected PasswordAuthentication getPasswordAuthentication() {
             }
             Transport.send(message);
         } catch (MessagingException e) {
-            logger.error("Sending mail to {} failed:\n{}", toAddresses, e.getMessage());
+            logger.error("Sending mail failed:\n{}", e.getMessage());
```
Comment (ivard) [inline src/main/java/foundation/privacybydesign/email/EmailSender.java:82]: And probably here too.

File: src/main/java/foundation/privacybydesign/email/EmailRestApi.java (hunk 1 above)
Comment (sanderhollaar) [reply src/main/java/foundation/privacybydesign/email/EmailRestApi.java:128]: verified: in the message of the exception `emailAddress` is not metioned

File: src/main/java/foundation/privacybydesign/email/EmailSender.java (hunk 2 above)
Comment (sanderhollaar) [reply src/main/java/foundation/privacybydesign/email/EmailSender.java:82]: also verified, thanks :-)

File: src/main/java/foundation/privacybydesign/email/EmailRestApi.java [hunk 3]
```diff
@@ -74,7 +74,7 @@ public Response sendEmail(@FormParam("email") String email,
                     url
             );
         } catch (AddressException e) {
-            logger.error("Invalid address: {}: {}", email, e.getMessage());
+            logger.error("Invalid address: {}", e.getMessage());
```
[the remark this answers is not in the document]
Comment (sanderhollaar) [reply src/main/java/foundation/privacybydesign/email/EmailRestApi.java:77]: email with space:

`ERROR foundation.privacybydesign.email.EmailRestApi - Invalid address: Local address contains control or whitespace`

**Review by ivard — APPROVED** [review]

**Pull request merged**